test_util/cluster.py

```diff
@@ -0,0 +1,292 @@
+import itertools
+import logging
+import os
+import stat
+from contextlib import contextmanager
+
+import passlib.hash
+import pkg_resources
+from retrying import retry
+
+import gen.calc
+import test_util.installer_api_test
+import test_util.test_runner
+from ssh.ssh_tunnel import SSHTunnel, TunnelCollection
+
+
+class Ssher:
+
+    def __init__(self, user, home_dir, key_path):
+        self.user = user
+        self.home_dir = home_dir
+        self.key_path = key_path
+
+        try:
+            os.chmod(self.key_path, stat.S_IREAD | stat.S_IWRITE)
+        except FileNotFoundError as exc:
+            raise Exception('SSH key not found at path {}'.format(self.key_path)) from exc
+        except OSError as exc:
+            raise Exception('Unable to set permissions on SSH key at {}: {}'.format(
+                self.key_path,
+                exc,
+            )) from exc
+
+    def tunnel(self, address):
+        return SSHTunnel(self.user, self.key_path, address)
+
+    @contextmanager
+    def tunnels(self, addresses):
+        hostports = [a + ':22' for a in addresses]
```

Comment:
`TunnelCollection` expects it in case `SSHTunnel` wants it: https://github.com/dcos/dcos/blob/38be502429c5ea93bdd4a8bc7a1916a66f786609/ssh/ssh_tunnel.py#L115 I'm not crazy about that API, but I'm hoping to limit this PR's spread to yet another module.